plugins/destination/postgresql/client/types.go

```diff
@@ -106,76 +106,73 @@ func (c *Client) PgToSchemaType(t string) schema.ValueType {
 	}
 }
 
-func (*Client) Pg10ToSchemaType(t string) schema.ValueType {
+func (c *Client) Pg10ToSchemaType(t string) arrow.DataType {
 	if strings.HasPrefix(t, "timestamp") {
-		return schema.TypeTimestamp
+		return arrow.FixedWidthTypes.Timestamp_us
+	}
+	if strings.HasSuffix(t, "[]") {
+		return arrow.ListOf(c.Pg10ToSchemaType(strings.TrimSuffix(t, "[]")))
 	}
 
 	switch t {
 	case "boolean":
-		return schema.TypeBool
-	case "bigint", "integer", "bigserial", "smallint", "smallserial", "serial":
-		return schema.TypeInt
-	case "double precision", "float", "real", "numeric":
-		return schema.TypeFloat
+		return arrow.FixedWidthTypes.Boolean
+	case "smallint":
+		return arrow.PrimitiveTypes.Int16
+	case "integer":
+		return arrow.PrimitiveTypes.Int32
+	case "bigint":
+		return arrow.PrimitiveTypes.Int64
+	case "real":
+		return arrow.PrimitiveTypes.Float32
+	case "double precision":
+		return arrow.PrimitiveTypes.Float64
 	case "uuid":
-		return schema.TypeUUID
+		return types.ExtensionTypes.UUID
 	case "bytea":
-		return schema.TypeByteArray
-	case "text[]":
-		return schema.TypeStringArray
+		return arrow.BinaryTypes.Binary
```

Comment:
Do we want to use `LargeBinary` here? I think we use it in SQLite.
Not sure if it makes a difference